docs/basics/101-139-hostingservices.rst

```diff
@@ -160,7 +162,163 @@ Should you employ GitHub workflows, for example, a token without "workflow" scop
+   configure-sibling(ok): . (sibling)
+   action summary:
+     configure-sibling (ok: 1)
+     create_sibling_gitlab (ok: 1)
+
+.. figure:: ../artwork/src/gitlab-layout-flat.png
+
+For a **collection** layout, the ``--project`` parameter determines the group name:
+
+.. code-block:: bash
+
+   $ datalad create-sibling-gitlab --site gitlab-general --layout collection --project my-datalad-root-level-group/this-will-be-the-group-name
+    create_sibling_gitlab(ok): . (dataset) [sibling repository 'gitlab' created at https://gitlab.com/my-datalad-root-level-group/this-will-be-the-group-name/project]
+    configure-sibling(ok): . (sibling)
+    action summary:
+      configure-sibling (ok: 1)
+      create_sibling_gitlab (ok: 1)
+
+.. figure:: ../artwork/src/gitlab-layout-collection.png
+
+Publishing datasets recursively
+"""""""""""""""""""""""""""""""
+
+When publishing a series of datasets recursively, the ``--project`` argument can not be used anymore - otherwise, all datasets in the hierarchy would attempt to create the same group or project over and over again.
+Instead, one configures the root level dataset, and the names for underlying datasets will be derived from this configuration:
+
+.. code-block::
+
+   # do the configuration for the top-most dataset
+   # either configure with Git
+   $ git config --local --replace-all \
+     datalad.gitlab-<gitlab-site>-project \
+     'my-datalad-root-level-group/DataLad-101_flat'
+   # or configure with DataLad
+   $ datalad configuration set \
+     datalad.gitlab-<gitlab-site>-project='my-datalad-root-level-group/DataLad-101_flat'
+
+Afterwards, publish dataset hierarchies with the ``--recursive`` flag:
+
+.. code-block:: bash
+
+   $ datalad create-sibling-gitlab --site gitlab-general --recursive --layout flat
+   create_sibling_gitlab(ok): . (dataset) [sibling repository 'gitlab' created at https://gitlab.com/my-datalad-root-level-group/DataLad-101_flat]
+   configure-sibling(ok): . (sibling)
+   create_sibling_gitlab(ok): midterm_project (dataset) [sibling repository 'gitlab' created at https://gitlab.com/my-datalad-root-level-group/DataLad-101_flat-midterm_project]
+   configure-sibling(ok): . (sibling)
+   create_sibling_gitlab(ok): midterm_project/input (dataset) [sibling repository 'gitlab' created at https://gitlab.com/my-datalad-root-level-group/DataLad-101_flat-midterm_project-input]
+   configure-sibling(ok): . (sibling)
+   create_sibling_gitlab(ok): recordings/longnow (dataset) [sibling repository 'gitlab' created at https://gitlab.com/my-datalad-root-level-group/DataLad-101_flat-recordings-longnow]
+   configure-sibling(ok): . (sibling)
+   action summary:
+     configure-sibling (ok: 4)
+     create_sibling_gitlab (ok: 4)
+
+Final step: Pushing to GitLab
+"""""""""""""""""""""""""""""
+
+Once you have set up your dataset sibling(s), you can push individual datasets or recursive across a hierarchy with ``datalad push --to gitlab``.
```

Comment:
```suggestion
Once you have set up your dataset sibling(s), you can push individual datasets with ``datalad push --to gitlab`` or push recursively across a hierarchy by adding the ``--recursive`` flag to the push command. 
```

Should we add the flag just to be clear?

Reply:
yes, makes sense - thanks for the suggestion!